File: backjoon/2004.py
n, m = map(int, input().split())
def count_t(n):
    answer = 0
    i = 2
    while i <= n:
        answer += n//i
        i = i*2
    return answer

def count_f(n):
    answer = 0
    i = 5
    while i <= n:
        answer += n//i
        i = i*5
    return answer
two = 0
five = 0
two += count_t(n)
five += count_f(n)
two -= count_t(m)
five -= count_f(m)
two -= count_t(n-m)
five -= count_f(n-m)

print(min(two, five))
print("|\\_/|\n|q p|   /}\n( 0 )\"\"\"\\\n|\"^\"`    |\n||_/=\\\\__|")

File: baekjoon/silver/2004/2004.go
package main

import "fmt"

func main() {
	var n, m int
	fmt.Scan(&n, &m)
	fmt.Println(min(countT(n)-countT(m)-countT(n-m), countF(n)-countF(m)-countF(n-m)))
}

func countT(n int) int {
	ans := 0
	i := 2
	for i <= n {
		ans += n / i
		i *= 2
	}
	return ans
}

func countF(n int) int {
	ans := 0
	i := 5
	for i <= n {
		ans += n / i
		i *= 5
	}
	return ans
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}
